src/rust/ide/view/src/project.rs

```diff
@@ -236,6 +230,48 @@ impl Model {
         node_id
     }
 
+    /// Ads a new node below `parent` and returns its ID. If there is not enough space right below
+    /// `parent` then the new node is moved to the right to first gap that is large enough.
+    pub fn add_node_below(&self, parent:NodeId) -> NodeId {
+        let parent_pos = self.graph_editor.model.get_node_position(parent).unwrap_or_default();
+        let styles     = StyleWatch::new(&self.app.display.scene().style_sheet);
+        use ensogl_theme::project as theme;
+        let x_gap    = styles.get_number(theme::default_x_gap_between_nodes);
+        let y_gap    = styles.get_number(theme::default_y_gap_between_nodes);
+        let y_offset = y_gap + node::HEIGHT;
+        let mut x    = parent_pos.x;
+        let y        = parent_pos.y - y_offset;
+        // Push x to the right until we find a position where we have enough space for the new
+        // node, including a margin of size `x_gap`/`y_gap` on all sides.
+        {
+            let nodes             = self.graph_editor.model.nodes.all.raw.borrow();
+            // We subtract `f32::EPSILON` to be sure that we do not include the selected node.
```

Comment:
I do not understand neither this comment nor this line of code. What would happen if we do not subtract it? Please describe it more clear in the docs :)

Reply:
What could happen is that rounding errors lead us to consider the parent root as overlapping, because lies exactly on the threshold where we should not count it.